apps/meteor/client/components/RoomIcon/OmnichannelRoomIcon/provider/OmnichannelRoomIconProvider.tsx

```diff
@@ -24,34 +24,55 @@ export const OmnichannelRoomIconProvider: FC = ({ children }) => {
 
 	return (
 		<OmnichannelRoomIconContext.Provider
-			value={useMemo(
-				() => ({
+			value={useMemo(() => {
+				const extractSnapshot = (app: string, iconName: string): AsyncState<string> => {
+					const icon = OmnichannelRoomIcon.get(app, iconName);
+
+					if (icon) {
+						return {
+							phase: AsyncStatePhase.RESOLVED,
+							value: icon,
+							error: undefined,
+						};
+					}
+
+					return {
+						phase: AsyncStatePhase.LOADING,
+						value: undefined,
```

Comment:
It's really necessary to send undefined as a value?

Reply:
It is:

```ts
export type AsyncState<T> =
	| { phase: AsyncStatePhase.LOADING; value: undefined; error: undefined }
	| { phase: AsyncStatePhase.LOADING; value: T; error: undefined }
	| { phase: AsyncStatePhase.LOADING; value: undefined; error: Error }
	| { phase: AsyncStatePhase.RESOLVED; value: T; error: undefined }
	| { phase: AsyncStatePhase.UPDATING; value: T; error: undefined }
	| { phase: AsyncStatePhase.REJECTED; value: undefined; error: Error };
```